### PR TITLE
Add shell and dilateXY

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: algorithm/cgal/Shell.h [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    double edge_size = 1;
+
+    Mesh_criteria criteria(p::facet_angle = angle, p::facet_size = sizing,
+                           p::facet_distance = approx,
+                           p::facet_topology = topology,
+                           p::edge_size = edge_size);
+
+    bool maybe_hollow = inner_offset > -rad2;
+
+    if (protect) {
```
Comment (BarbourSmith) [inline algorithm/cgal/Shell.h:159]: 👍

File: api/shape/cut.js [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
+    args,
+    'geometry',
+    'geometries',
+    'modes'
+  );
+  return Shape.fromGeometry(
+    cutGeometry(
+      first,
+      rest,
+      modes.includes('open'),
```
Comment (BarbourSmith) [inline api/shape/cut.js:17]: What do these do?...Or rather when would the user use them?

**Review by BarbourSmith — APPROVED** [review]
Does this change seem right?

image

If that is OK it looks great to me

**Comment by pentacular** [comment]
> Does this change seem right?
> 
> image
> 
> If that is OK it looks great to me

I temporarily disabled toolpaths :( So it's expected for now.

File: api/shape/cut.js (hunk 2 above)
Comment (pentacular) [reply api/shape/cut.js:17]: The different between a.cut and a.Cut is if a is included implicitly in the cut, or just used as context.

a.cut(b, c) === a.Cut(self(), b, c)

One example where you might use a.Cut is a.Cut(offset(1), inset(1))

Comment (pentacular) [reply api/shape/cut.js:17]: This should be similarly true across the rest of the operations -- e.g., and, And, etc.

Comment (BarbourSmith) [reply api/shape/cut.js:17]: That makes a lot of sense...nice!

**Pull request merged**